ddl/cluster.go

```diff
@@ -140,13 +143,85 @@ func checkFlashbackCluster(w *worker, d *ddlCtx, t *meta.Meta, job *model.Job, f
 	return nil
 }
 
+func addToSlice(schema string, tableName string, tableID int64, nonFlashbackIDs, allIDs []int64) ([]int64, []int64) {
+	if filter.IsSystemSchema(schema) && !strings.HasPrefix(tableName, "stats_") {
+		nonFlashbackIDs = append(nonFlashbackIDs, tableID)
+	}
+	allIDs = append(allIDs, tableID)
+	return nonFlashbackIDs, allIDs
+}
+
+// GetFlashbackKeyRanges get keyRanges for flashback cluster
+func GetFlashbackKeyRanges(sess sessionctx.Context, startKey kv.Key) ([]kv.KeyRange, error) {
+	schemas := sess.GetDomainInfoSchema().(infoschema.InfoSchema).AllSchemas()
+
+	var keyRanges []kv.KeyRange
+	var allPhysicalIDs, nonFlashbackPhysicalIDs []int64
```

Comment:
IMHO it's a good idea to add some comments with detailed description and examples on the logic here:

* The semantic of `allPhysicalIDs` and `nonFlashbackPhysicalIDs`(input).
* The semantic of `keyRanges`(output).
* The algorithm used to make ranges efficiently for the cases when there are quite a lot of tables -- O(n * logn) of time complexity.

Reply:
Updated, PTAL @bb7133